ChromaDB & Testing

mwetzka03 opened the pull request:
Ich habe die ChromaDB initialisiert mit und ohne Langchain (ohne weil Befehle wie delete funktionieren sonst nicht). Außerdem habe ich einen Test geschrieben für die API Befehle der DB.

mwetzka03 committed:
ChromaDB & Testing

Ich habe die ChromaDB initialisiert mit und ohne Langchain (ohne weil Befehle wie delete funktionieren sonst nicht). Außerdem habe ich einen Test geschrieben für die API Befehle der DB.

mwetzka03 committed:
Update LangchainClient.py

PaulNeuselDHBW commented on app/core/services/DatabaseService.py:
```diff
@@ -0,0 +1,27 @@
+    def add_documents(self, texts, metadatas=None, ids=None):
+        self.db.add(texts, metadatas, ids)
+
+    def search_documents(self, query, k=3):
```
What is the difference to get? query -> not id?

mwetzka03 replied:
ein Dokument per id oder mehrere Dokumente

PaulNeuselDHBW commented on app/core/services/DatabaseService.py:
```diff
@@ -0,0 +1,27 @@
+    to an implementation of the DatabasePort interface.
+
+    This service acts as the application's main entry point for storing,
+    retrieving, updating, deleting, and searching documents.
```
Please add the needed format of the text input for the database.
As DTO (Data Transfer Object) or optional as example comment.

PaulNeuselDHBW replied:
See issue #43 .

PaulNeuselDHBW commented on app/infrastructure/db/DatabaseAdapter.py:
```diff
@@ -0,0 +1,32 @@
+    to the LangchainClient, which provides access to both semantic search and
+    full CRUD operations using ChromaDB.
+
+    It acts as the bridge between your core logic and the infrastructure layer.
```
Normally this just implements the Port by calling the methods of the ChromaDatabase.
I am not quite sure about what the difference is between the Chroma interaction and this langchain Client.
Seems like we only need one?!

PaulNeuselDHBW replied:
Resolved in another Comment!

PaulNeuselDHBW commented on app/infrastructure/rag/langchain/LangchainClient.py:
```diff
@@ -0,0 +1,55 @@
+from langchain_huggingface import HuggingFaceEmbeddings
+from chromadb import PersistentClient
+
+class LangchainClient:
```
This is perfect, but see #43 and #42 .
I would recommend marking it as a service, but this my personal preference depending on the architecture.

PaulNeuselDHBW replied:
My comment makes no sense.

PaulNeuselDHBW commented on app/presentation/controller/RESTController.py:
```diff
@@ -1,19 +1,69 @@
-#from app.core.services.RAGService
-#from flask_restful import Api, Resource
+from flask import Blueprint, request, jsonify
+from app.core.services.DatabaseService import DatabaseService
```
This REST-Points would have been another ticket ;)
You are way to fast xD

mwetzka03 replied:
wenn man das nicht implementiert kann man nicht schauen obs funktioniert

PaulNeuselDHBW reviewed (dismissed):
Habe ein paar kleinere Kommentare hinzugefügt, sollte aber im großen und Ganzen passen. Sollten diese eingepflegt sein, würde ich die Anwendung nochmal abschließend testen und das Ticket ist done.
Wenn möglich nur eine kurze Erklärung in der ReadMe beifügen :)

PaulNeuselDHBW commented on app/presentation/controller/RESTController.py:
```diff
@@ -1,19 +1,69 @@
 
-#Instanz der Flask-Klasse
 rest_bp = Blueprint("rest", __name__)
+rag_service = DatabaseService(DatabaseAdapter())
```
edit : Name ist eventuell etwas unpassend?!

mwetzka03 committed:
test !nicht funktional

mwetzka03 committed:
Fixes & Implementierung

Fehler bei den Tests behoben, DocumentDTO eingeführt, Embedding hinzugefügt, Score System bei Similiarity Search

mwetzka03 committed:
ChromaDB Leerung

mwetzka03 commented:
Ich habe Neuerungen eingemacht, sollte jetzt passen

karatemxdchen reviewed (approved):
Ist gut
